lib/cocoapods/installer/analyzer.rb

```diff
@@ -453,6 +453,17 @@ def generate_targets(resolver_specs_by_target, target_inspections)
           aggregate_target.pod_targets.each(&:mark_application_extension_api_only)
         end
 
+        aggregate_targets.each do |aggregate_target|
```

Comment:
we are already looping above all targets, can you combine that into the single loop, might have to remove the `next unless is_app_extension` line etc etc.